### PR TITLE
semantic-ui: Use getDisplayLabel to properly show labels for widgets

### PR DESCRIPTION
**Pull request by jacqueswho**
### Reasons for making this change

Features:
- implement getDisplayLabel from core/utils

https://github.com/rjsf-team/react-jsonschema-form/issues/1946

### Checklist

* [x] **I'm updating documentation**
  - [x] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests
  - [x] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
* [x] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: packages/semantic-ui/package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "test": "cross-env NODE_ENV=test BABEL_ENV=test jest"
   },
   "peerDependencies": {
-    "@rjsf/core": "^2.0.0",
+    "@rjsf/core": "2.4.2",
```
Comment (epicfaace) [inline packages/semantic-ui/package.json:31]: Was this intentional? I think this code is still compatible with @rjsf/core 2.0.0-2.4.2?

File: packages/core/src/utils.js [hunk 2]
```diff
@@ -364,11 +364,13 @@ export function mergeDefaultsWithFormData(defaults, formData) {
 
 export function getUiOptions(uiSchema) {
   // get all passed options from ui:widget, ui:options, and ui:<optionName>
+  if (!uiSchema) {
```
Comment (epicfaace) [inline packages/core/src/utils.js:367]: See my comments in #2226 -- I think these changes in this file might not actually be necessary. If they're not necessary, can we remove them?

File: packages/semantic-ui/package.json (hunk 1 above)
Comment (jacqueswho) [reply packages/semantic-ui/package.json:31]: it was to make sure I am on the latest core.

File: packages/core/src/utils.js (hunk 2 above)
Comment (jacqueswho) [reply packages/core/src/utils.js:367]: i needed this change for the render tests to pass. I linked core, made the change and they passed

**Comment by ErQrYfkrju** [comment]
@agustin107 ping

**Comment by ErQrYfkrju** [comment]
Hello @agustin107 @epicfaace @jacqueswho any news on this pull request?

**Comment by jacqueswho** [comment]
> Hello @agustin107 @epicfaace @jacqueswho any news on this pull request?

sorry @ErQrYfkrju cant do anything. just waiting.

File: packages/semantic-ui/src/CheckboxesWidget/CheckboxesWidget.js [hunk 3]
```diff
@@ -16,19 +16,20 @@ function deselectValue(value, selected) {
...
-  autofocus,
-  readonly,
-  onChange,
-  onBlur,
-  onFocus,
-  formContext,
-  schema,
-}) {
+function CheckboxesWidget(props) {
+  const {
```
Comment (epicfaace) [inline packages/semantic-ui/src/CheckboxesWidget/CheckboxesWidget.js:20]: Let's undo this change, unless you think it's better to have it this way?

File: packages/semantic-ui/src/PasswordWidget/PasswordWidget.js [hunk 4]
```diff
@@ -3,35 +3,42 @@ import React from "react";
...
-  onFocus,
-  autofocus,
-  options,
-  schema,
-  formContext,
-}) {
+import {  utils } from "@rjsf/core";
+const { getDisplayLabel } = utils;
+function PasswordWidget(props) {
+  const {
```
Comment (epicfaace) [inline packages/semantic-ui/src/PasswordWidget/PasswordWidget.js:9]: Let's undo this change (props destructuring style)

File: packages/core/src/utils.js [hunk 5]
```diff
@@ -394,13 +393,14 @@ export function getDisplayLabel(schema, uiSchema, rootSchema) {
...
   }
+
   if (schema.type === "object") {
     displayLabel = false;
   }
   if (schema.type === "boolean" && !uiSchema["ui:widget"]) {
     displayLabel = false;
   }
-  if (uiSchema["ui:field"]) {
+  if (uiSchema && uiSchema["ui:field"]) {
```
Comment (epicfaace) [inline packages/core/src/utils.js:403]: Can you explain again why this change was needed? I think we might have discussed this earlier, but I don't remember what we discussed earlier. If possible, we should remove it, but if we need it, we should add a test that covers it.

File: packages/semantic-ui/src/CheckboxWidget/CheckboxWidget.js [hunk 6]
```diff
@@ -17,8 +16,10 @@ function CheckboxWidget(props) {
     options,
     onFocus,
     formContext,
+    schema,
   } = props;
   const semanticProps = getSemanticProps({ formContext, options });
+  const desc = label || schema.description;
```
Comment (epicfaace) [inline packages/semantic-ui/src/CheckboxWidget/CheckboxWidget.js:22]: Can we undo this change for this PR for now?

Normally, if the label and description are both specified, we'd want to show both instead of just overriding the description with the label (https://github.com/rjsf-team/react-jsonschema-form/blob/b78d9ef280eddf5bda0f97e5a3445c6a1fd35c99/packages/core/src/components/widgets/CheckboxWidget.js#L27-L42). However, I think that might be beyond the scope of this PR and we can make another issue / PR addressing it.

File: packages/semantic-ui/src/RangeWidget/RangeWidget.js [hunk 7]
```diff
@@ -6,21 +6,21 @@ import { utils } from '@rjsf/core';
...
-  options,
-  schema,
-  onChange,
-  required,
-  label,
-  name,
-  id,
-  formContext,
-}) {
+function RangeWidget(props) {
```
Comment (epicfaace) [inline packages/semantic-ui/src/RangeWidget/RangeWidget.js:9]: Let's undo this change

File: packages/semantic-ui/src/SelectWidget/SelectWidget.js [hunk 8]
```diff
@@ -59,22 +60,24 @@ const processValue = (schema, value) => {
...
-  value,
-  multiple,
-  placeholder,
-  autofocus,
-  onChange,
-  onBlur,
-  onFocus,
-}) {
+function SelectWidget(props) {
+  const {
```
Comment (epicfaace) [inline packages/semantic-ui/src/SelectWidget/SelectWidget.js:64]: Let's undo this change

File: packages/core/src/utils.js [hunk 9]
```diff
@@ -1,9 +1,9 @@
-import React from "react";
-import * as ReactIs from "react-is";
-import mergeAllOf from "json-schema-merge-allof";
 import fill from "core-js/library/fn/array/fill";
-import union from "lodash/union";
+import mergeAllOf from "json-schema-merge-allof";
 import jsonpointer from "jsonpointer";
+import union from "lodash/union";
```
Comment (epicfaace) [inline packages/core/src/utils.js:4]: Undo change in order of the imports

File: packages/semantic-ui/src/TextareaWidget/TextareaWidget.js [hunk 10]
```diff
@@ -3,36 +3,43 @@ import React from "react";
...
-  onFocus,
-  onChange,
-  options,
-  schema,
-  formContext,
-}) {
+import {  utils } from "@rjsf/core";
+const { getDisplayLabel } = utils;
+function TextareaWidget(props) {
+  const {
```
Comment (epicfaace) [inline packages/semantic-ui/src/TextareaWidget/TextareaWidget.js:9]: Undo props change

File: packages/semantic-ui/src/UpDownWidget/UpDownWidget.js [hunk 11]
```diff
@@ -4,28 +4,36 @@ import PropTypes from "prop-types";
...
-  onBlur,
-  onFocus,
-  autofocus,
-  rawErrors,
-  formContext,
-}) {
+import {  utils } from "@rjsf/core";
+const { getDisplayLabel } = utils;
+function UpDownWidget(props) {
+  const {
```
Comment (epicfaace) [inline packages/semantic-ui/src/UpDownWidget/UpDownWidget.js:10]: Undo props change

**Review by epicfaace — COMMENTED** [review]
Hey @jacqueswho , just a few things that we need before getting this PR ready:
- can you remove the extraneous changes that aren't really relevant for this change? That will ensure this PR is narrowly scoped to only include the get display util-related changes. For example, changes in props destructuring, extra blank lines deleted / added, etc.
- let's add some tests with the different schemas / uischemas with labels to ensure that all the cases here are tested by these tests so that they don't regress in the future.

File: packages/semantic-ui/src/CheckboxesWidget/CheckboxesWidget.js (hunk 3 above)
Comment (jacqueswho) [reply packages/semantic-ui/src/CheckboxesWidget/CheckboxesWidget.js:20]: i prefer this approach, as the other themes do it this way as well, so they can perform functions on all props, or validate props and so forth

File: packages/core/src/utils.js (hunk 9 above)
Comment (jacqueswho) [reply packages/core/src/utils.js]: sorry I have auto save sort imports

**Comment by jacqueswho** [comment]
@epicfaace have made all the changes. except please keep the props approach, as the in other pull requests it is like this, and most of the other themes , use this approach to make it easier for applying validation and functions for all props

**Review by epicfaace — COMMENTED** [review]
Thanks, can you update the snapshots so the tests pass @jacqueswho ?

**Comment by jacqueswho** [comment]
> Thanks, can you update the snapshots so the tests pass @jacqueswho ?

Hi @epicfaace  if you look at the files changed I already did. and the tests pass for me, but my version is  @rjsf/semantic-ui@2.5.0 but not sure why the tests are running 2.5.1?

**Comment by epicfaace** [comment]
@jacqueswho hmm, I wonder if you actually need to revert the snapshot changes in this PR. If you run `npm test -- -u` then do the snapshots revert or do they still stay the same?

**Comment by jacqueswho** [comment]
@epicfaace I ran that command and no changes to push?

**Comment by jacqueswho** [comment]
@epicfaace please can this be merged

File: packages/semantic-ui/src/SelectWidget/SelectWidget.js [hunk 12]
```diff
@@ -94,10 +97,12 @@ function SelectWidget({
     // eslint-disable-next-line no-shadow
     target: { value },
   }) => onFocus && onFocus(id, processValue(schema, value));
+
   return (
     <Form.Dropdown
       key={id}
       name={name}
+      label={label || schema.title}
```
Comment (epicfaace) [inline packages/semantic-ui/src/SelectWidget/SelectWidget.js:105]: Shouldn't we use getDisplayLabel here as well?

File: packages/semantic-ui/src/TextWidget/TextWidget.js [hunk 13]
```diff
@@ -3,39 +3,46 @@ import React from "react";
...
-  schema,
-  value,
-  onChange,
-  onBlur,
-  onFocus,
-  autofocus,
-  options,
-  formContext,
-}) {
+import {  utils } from "@rjsf/core";
```
Comment (epicfaace) [inline packages/semantic-ui/src/TextWidget/TextWidget.js:6]:
```suggestion
import { utils } from "@rjsf/core";
```

**Review by epicfaace — COMMENTED** [review]
Sorry @jacqueswho for the delay, only now getting a better chance to look at this. In addition to the changes I mentioned, can you also add a simple snapshot test that checks that https://github.com/rjsf-team/react-jsonschema-form/issues/1946 is working for a simple example?

File: packages/semantic-ui/src/SelectWidget/SelectWidget.js (hunk 12 above)
Comment (jacqueswho) [reply packages/semantic-ui/src/SelectWidget/SelectWidget.js:105]: I don't think so, as I have seen all other themes not do this? must be a reason why, unless you see otherwise I can do the same

Comment (epicfaace) [reply packages/semantic-ui/src/SelectWidget/SelectWidget.js:105]: It's probably a mistake, I think all widgets should be using getDisplayLabel even in other themes. For now, though, let's just leave it as is for consistency.

**Review by epicfaace — APPROVED** [review]
